11-payment-encoding.md

```diff
@@ -0,0 +1,355 @@
+# BOLT #11: Invoice Protocol for Lightning Payments
+
+A simple, extensible QR-code-ready protocol for requesting payments
+over Lightning.
+
+# Table of Contents
+  
+# Encoding Overview
+
+The format for a lightning invoice uses
+[bech32 encoding](https://github.com/bitcoin/bips/blob/master/bip-0173.mediawiki),
+which is already proposed for bitcoin Segregated Witness, and can be
+simply reused here even though its 6-character checksum is optimized
+for manual entry, which is unlikely to happen often given the length
+of lightning invoices.
+
+## Requirements
+
+A writer MUST encode the the payment request in Bech32 as specified in
+BIP-0173.  A reader MUST parse the address as Bech32 as specified in
+BIP-0173, and MUST fail if the checksum is incorrect.
+
+# Human Readable Part
+
+The human readable part consists of two sections:
+1. `prefix`: `ln` + BIP-0173 currency prefix (e.g. `lnbc`, `lntb`)
+1. `amount`: optional number in that currency, followed by optional
+   `multiplier`.
+
+The following `multiplier` letters are defined:
+
+* `m` (milli): multiply by 0.001
+* `u` (micro): multiply by 0.000001
+* `n` (nano): multiply by 0.000000001
+* `p` (pico): multiply by 0.000000000001
+
+## Requirements
+
+A writer MUST include `amount` if payments will be refused if less
+than that.  A writer MUST encode `amount` as a positive decimal
+integer with no leading zeroes, SHOULD use the shortest representation
+possible.
+
+A reader MUST fail if it does not understand the `prefix`.  A reader
+SHOULD fail if `amount` contains a non-digit, or is followed by
+anything except a `multiplier` in the table above.
+
+A reader SHOULD indicate if amount is unspecified, otherwise it MUST
+multiply `amount` by the `multiplier` value (if any) to derive the
+amount required for payment.
+
+## Rationale
+
+The `amount` is encoded into the human readable part, as it's fairly
+readable and a useful indicator of how much is being requested.
+
+Donation addresses often don't have an associated amount, so `amount`
+is optional in that case: usually a minimum payment is required for
+whatever is being offered in return.
+
+# Data Part
+
+The data part consists of multiple sections:
+
+1. `timestamp`: seconds-since-1970 (35 bits, big-endian)
+1. Zero or more tagged parts.
+1. `signature`: bitcoin-style signature of above. (520 bits)
+
+## Requirements
+
+A writer MUST set `timestamp` to the time to
+the number of seconds since Midnight 1 January 1970, UTC in
+big-endian.  A writer MUST set `signature` to a valid
+512-bit secp256k1 signature of the double SHA2 256-bit hash of the
+Human Readable Part concatenated with a byte for each 5 bits of the
+Data Part, with a trailing byte containing the recovery ID (0, 1, 2 or
```

Comment:
This is a very good q!  It's currently defined by the secp256k1 implementation, and I can't find a description of it anywhere else.  I think we need to define the values here in this case.